File: frontend/services/assistant.ts
import axios from "axios";

import { api } from "./api";


export type AssistantResponse = {
    answer: string;
};


export async function askAssistant(
    question: string
): Promise<AssistantResponse> {

    try {

        const response = await api.post<AssistantResponse>(
            "/assistant/",
            {
                question,
            }
        );

        return response.data;

    } catch (error) {

        if (axios.isAxiosError(error)) {

            throw new Error(
                error.response?.data?.detail
                ?? "AI assistant is unavailable."
            );

        }

        throw error;

    }

}

File: backend/app/api/assistant.py
from fastapi import (
    APIRouter,
    Depends,
    HTTPException
)
from sqlalchemy.orm import Session

from app.api.dependencies import (
    get_current_staff_user
)
from app.models.assistant_model import (
    AssistantRequest,
    AssistantResponse
)
from app.database.database import get_db


router = APIRouter(
    prefix="/assistant",
    tags=["AI HR Assistant"]
)


def ask_assistant(
    question: str,
    db: Session
):

    from app.services.assistant_service import (
        ask_assistant as service
    )

    return service(question, db=db)


@router.post(
    "/",
    dependencies=[
        Depends(get_current_staff_user)
    ],
    response_model=AssistantResponse
)
def assistant_chat(
    request: AssistantRequest,
    db: Session = Depends(get_db)
):
    """
    Ask the AI HR assistant about indexed resumes.
    """

    try:

        answer = ask_assistant(
            request.question,
            db
        )

    except Exception:

        raise HTTPException(
            status_code=500,
            detail="Assistant is currently unavailable."
        )

    return AssistantResponse(
        answer=answer
    )
